Entities/Materials/MaterialCommon.as

```diff
@@ -172,15 +172,32 @@ namespace Material
   }
 
   // Server-side: Create material from a blob
-  void fromBlob(CBlob@ this, CBlob@ blob, float &in damage)
+  void fromBlob(CBlob@ this, CBlob@ blob, float &in damage, CBlob@ damageBlob = null)
   {
     if (damage <= 0.f) return;
 
     // Return unless it's a harvest blob
     if (not blob.exists('harvest')) return;
-
     dictionary harvest;
     blob.get('harvest', harvest);
+    u16 maxHarvest = -1;//keep -1 unless we want a cap
+
+    if(damageBlob !is null)
+    {
+      //if we are going to have more blobs, move to switch
```

Comment:
Would be great if the indentation here was kag style (tabs, equivalent to 4 wide spaces) - i know editing in github makes this annoying though.